### PR TITLE
Fixes #2277

### LINKED ISSUE #2277: Failure when parsing mol block with M  PXA
**Description:**

`M  PXA` indicates a "Phantom Extra Atom". These show up sometimes in places like ChEMBL structures for peptides and lead to parse failures:

```
In [7]: Chem.MolFromMolBlock(''' 
   ...:   Mrv1810 02151911552D           
   ...:  
   ...:  13 12  0  0  1  0            999 V2000 
   ...:    -3.6588  -26.0592    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0 
   ...:    -2.9453  -27.2971    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9453  -26.4713    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -3.6588  -25.2360    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9467  -24.8200    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9467  -23.9968    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.2304  -25.2358    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -1.5102  -26.4716    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.7989  -25.2306    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.7989  -26.0582    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -4.3730  -26.4732    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.2277  -26.0635    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.0839  -26.4708    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:   1  3  1  0  0  0  0 
   ...:   3  2  2  0  0  0  0 
   ...:   1  4  1  1  0  0  0 
   ...:   4  5  1  0  0  0  0 
   ...:   5  6  1  0  0  0  0 
   ...:   5  7  1  0  0  0  0 
   ...:   8 10  1  0  0  0  0 
   ...:  10  9  2  0  0  0  0 
   ...:   3 12  1  0  0  0  0 
   ...:  12  8  1  0  0  0  0 
   ...:  11  1  1  0  0  0  0 
   ...:  10 13  1  0  0  0  0 
   ...: M  PXA  11   -5.0817  -26.0408    0.0000 H 
   ...: M  END 
   ...: ''')                                                                                           

In [8]: Chem.MolFromMolBlock(''' 
   ...:   Mrv1810 02151911552D           
   ...:  
   ...:  13 12  0  0  1  0            999 V2000 
   ...:    -3.6588  -26.0592    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0 
   ...:    -2.9453  -27.2971    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9453  -26.4713    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -3.6588  -25.2360    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9467  -24.8200    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.9467  -23.9968    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.2304  -25.2358    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -1.5102  -26.4716    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.7989  -25.2306    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.7989  -26.0582    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -4.3730  -26.4732    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -2.2277  -26.0635    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:    -0.0839  -26.4708    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0 
   ...:   1  3  1  0  0  0  0 
   ...:   3  2  2  0  0  0  0 
   ...:   1  4  1  1  0  0  0 
   ...:   4  5  1  0  0  0  0 
   ...:   5  6  1  0  0  0  0 
   ...:   5  7  1  0  0  0  0 
   ...:   8 10  1  0  0  0  0 
   ...:  10  9  2  0  0  0  0 
   ...:   3 12  1  0  0  0  0 
   ...:  12  8  1  0  0  0  0 
   ...:  11  1  1  0  0  0  0 
   ...:  10 13  1  0  0  0  0 
   ...: M  END 
   ...: ''')                                                                                           
Out[8]: <rdkit.Chem.rdchem.Mol at 0x7f42b2597850>
```

The parsing for this was added as part of the S-groups PR.

### PR DESCRIPTION
**Pull request by greglandrum**
This adds a check to make sure that we generate a warning instead of an exception if an SGroup that isn't present is referenced.

### CODE REVIEW

**Comment by greglandrum** [comment]
@ricrogz : would be great if you could take a look at this too just to make sure I didn't screw anything up too badly

File: Code/GraphMol/FileParsers/MolSGroupParsing.cpp [hunk 1]
```diff
@@ -470,7 +535,16 @@ void ParseSGroupV2000PXALine(IDX_TO_SGROUP_MAP &sGroupMap, RWMol *mol,
   unsigned int pos = 6;
 
   unsigned int sgIdx = ParseSGroupIntField(text, line, pos);
-
+  if (sGroupMap.find(sgIdx) == sGroupMap.end()) {
+    BOOST_LOG(rdWarningLog) << "SGroup " << sgIdx << " referenced on line "
+                            << line << " not found." << std::endl;
+    return;
+  }
+  if (sGroupMap.find(sgIdx) == sGroupMap.end()) {
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/MolSGroupParsing.cpp]: The check is duplicated here.

File: Code/GraphMol/FileParsers/MolSGroupParsing.cpp [hunk 2]
```diff
@@ -515,6 +594,16 @@ void ParseSGroupV2000SCLLine(IDX_TO_SGROUP_MAP &sGroupMap, RWMol *mol,
   unsigned int pos = 6;
 
   unsigned int sgIdx = ParseSGroupIntField(text, line, pos);
+  if (sGroupMap.find(sgIdx) == sGroupMap.end()) {
+    BOOST_LOG(rdWarningLog) << "SGroup " << sgIdx << " referenced on line "
+                            << line << " not found." << std::endl;
+    return;
+  }
+  if (sGroupMap.find(sgIdx) == sGroupMap.end()) {
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/MolSGroupParsing.cpp]: Check duplicated here, too.

File: Code/GraphMol/FileParsers/MolSGroupParsing.cpp [hunk 3]
```diff
@@ -470,7 +535,16 @@ void ParseSGroupV2000PXALine(IDX_TO_SGROUP_MAP &sGroupMap, RWMol *mol,
   unsigned int pos = 6;
 
   unsigned int sgIdx = ParseSGroupIntField(text, line, pos);
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/MolSGroupParsing.cpp]: I made a mistake here when I implemented the SGroups: the leading number in the field's data is not an sgIdx, but an AtomIdx.

I didn't understand the spec on these were well, and I didn't have any sample files with PXA lines to help me understand them. So, I was actually not going to implement PXA, but looks that at some point I did, in the wrong way.

Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolSGroupParsing.cpp]: Ah, that explains it.
Ok, I updated the PR, moved that chunk of parsing code to `MolFileParser.cpp`, and now have it just set an atom property.

**Comment by greglandrum** [comment]
@ricrogz Right you are. Thanks for catching that.

File: Code/GraphMol/FileParsers/MolFileWriter.cpp [hunk 4]
```diff
@@ -1178,6 +1191,7 @@ std::string outputMolToMolBlock(const RWMol &tmol, int confId,
     res += GetMolFileZBOInfo(tmol);
 
     res += GetMolFileSGroupInfo(tmol);
+    res += GetMolFilePXAInfo(tmol);
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolFileWriter.cpp]: makes sense. I'll change it

**Comment by greglandrum** [comment]
Now part of #2283

**Pull request closed without merging**